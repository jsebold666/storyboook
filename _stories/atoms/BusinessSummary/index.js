import React from 'react';
import { number, text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import BusinessSummary from '../../../components/atoms/BusinessSummary';

const component = () => (
  <BusinessSummary
    onChange={action('changed')}
    totalValue={number('totalValue', 1)}
    anualChange={number('anualChange', -2)}
    labels={[
      text('label1', 'PRÊMIO R$'),
      text('label2', 'Total Ano'),
      text('label3', 'Ano passado'),
      text('label4', 'Mês passado')
    ]}
    monthlyChange={number('monthlyChange', 3)}
  />
);

export default [readme, component];
