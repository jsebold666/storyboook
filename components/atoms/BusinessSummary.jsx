import styled from 'styled-components';
import React from 'react';
import PropTypes from 'prop-types';
import shared from '../consts/shared';

import UpIcon from '../../static/images/arrow-percent-up.svg';
import DownIcon from '../../static/images/arrow-percent-down.svg';

const fontSize = 16;

const Wrapper = styled.div`
  display: flex;
  justify-content: space-around;
  margin-top: 20px;
  text-align: left;

  div {
    align-self: center;
  }

  .subtitle {
    font-size: 12px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.5;
    letter-spacing: normal;
    color: ${shared['bluish-grey']};
  }

  .upper-title {
    font-size: 12px;
    font-weight: 500;
    font-style: normal;
    font-stretch: normal;
    line-height: 2;
    letter-spacing: normal;
    color: ${shared['bluish-grey']};
  }

  .number {
    font-size: ${fontSize}px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.25;
    letter-spacing: normal;
    color: ${shared['slate']};
  }

  .up-or-down-icon {
    height: ${fontSize}px;
    padding-right: 12px;
  }
`;

export default function BusinessSummary(props) {
  return (
    <Wrapper>
      <SummaryItem
        uppperTitle={props.labels[0]}
        value={props.totalValue}
        subtitle={props.labels[1]}
        isCurrency
      />
      <SummaryItem uppperTitle="" value={props.anualChange} subtitle={props.labels[2]} />
      <SummaryItem uppperTitle="" value={props.monthlyChange} subtitle={props.labels[3]} />
    </Wrapper>
  );
}

function SummaryItem(props) {
  const { value } = props;
  const formatedValue = props.isCurrency ? brCurrency(value) : `${brNumber(value)} %`;
  const leftValueIcon = props.isCurrency ? null : renderIcon(value);
  return (
    <div>
      <span className="upper-title">{props.uppperTitle}</span>
      <br />
      <span className="number">
        {leftValueIcon}
        <span>{formatedValue}</span>
      </span>
      <br />
      <span className="subtitle">{props.subtitle}</span>
    </div>
  );
}

function renderIcon(value) {
  const convertedValue = parseFloat(value);
  const icon = convertedValue >= 0 ? <UpIcon /> : <DownIcon />;
  return <span className="up-or-down-icon">{icon}</span>;
}

function brCurrency(value) {
  if (value >= 1e9) {
    return brNumber(value / 1e9) + ' bi';
  } else if (value >= 1e6) {
    return brNumber(value / 1e6) + ' mi';
  } else if (value >= 1e3) {
    return brNumber(value / 1e3) + ' mil';
  }
  return value;
}

function brNumber(value) {
  if (typeof value === 'number') {
    return value.toFixed(2).replace('.', ',');
  } else if (typeof value === 'string') {
    return brNumber(parseFloat(value));
  }
  return 'N/A';
}
BusinessSummary.propTypes = {
  totalValue: PropTypes.number.isRequired,
  anualChange: PropTypes.number.isRequired,
  monthlyChange: PropTypes.number.isRequired,
  labels: PropTypes.arrayOf(PropTypes.string).isRequired
};
