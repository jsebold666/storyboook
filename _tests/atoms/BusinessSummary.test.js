/* globals mount */
import React from 'react';
import BusinessSummary from '../../components/atoms/BusinessSummary';

describe('Expect <BusinessSummary>', () => {
  it('to render', () => {
    const props = {
      totalValue: 1,
      anualChange: -2,
      monthlyChange: 3,
      labels: ['PRÊMIO R$', 'Total Ano', 'Ano passado', 'Mês passado'],
      onChange: () => {}
    };

    const wrapper = mount(<BusinessSummary {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
