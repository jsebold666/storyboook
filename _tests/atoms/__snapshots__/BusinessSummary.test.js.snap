// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <BusinessSummary> to render 1`] = `
<BusinessSummary
  anualChange={-2}
  labels={
    Array [
      "PRÊMIO R$",
      "Total Ano",
      "Ano passado",
      "Mês passado",
    ]
  }
  monthlyChange={3}
  onChange={[Function]}
  totalValue={1}
>
  <BusinessSummary__Wrapper>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "BusinessSummary__Wrapper-sc-1ce3iu1-0",
            "isStatic": true,
            "lastClassName": "iTggax",
            "rules": Array [
              "
  display: flex;
  justify-content: space-around;
  margin-top: 20px;
  text-align: left;

  div {
    align-self: center;
  }

  .subtitle {
    font-size: 12px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.5;
    letter-spacing: normal;
    color: ",
              "#859099",
              ";
  }

  .upper-title {
    font-size: 12px;
    font-weight: 500;
    font-style: normal;
    font-stretch: normal;
    line-height: 2;
    letter-spacing: normal;
    color: ",
              "#859099",
              ";
  }

  .number {
    font-size: ",
              "16",
              "px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.25;
    letter-spacing: normal;
    color: ",
              "#4a5365",
              ";
  }

  .up-or-down-icon {
    height: ",
              "16",
              "px;
    padding-right: 12px;
  }
",
            ],
          },
          "displayName": "BusinessSummary__Wrapper",
          "render": [Function],
          "styledComponentId": "BusinessSummary__Wrapper-sc-1ce3iu1-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="BusinessSummary__Wrapper-sc-1ce3iu1-0 iTggax"
      >
        <SummaryItem
          isCurrency={true}
          subtitle="Total Ano"
          uppperTitle="PRÊMIO R$"
          value={1}
        >
          <div>
            <span
              className="upper-title"
            >
              PRÊMIO R$
            </span>
            <br />
            <span
              className="number"
            >
              <span>
                1
              </span>
            </span>
            <br />
            <span
              className="subtitle"
            >
              Total Ano
            </span>
          </div>
        </SummaryItem>
        <SummaryItem
          subtitle="Ano passado"
          uppperTitle=""
          value={-2}
        >
          <div>
            <span
              className="upper-title"
            />
            <br />
            <span
              className="number"
            >
              <span
                className="up-or-down-icon"
              >
                <DownIcon
                  height="17"
                  viewBox="0 0 16 17"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    height="17"
                    viewBox="0 0 16 17"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14.5 8.5l-5.793 5.793a1 1 0 0 1-1.414 0L1.5 8.5m6.5-7v13"
                      fill="none"
                      stroke="#F66"
                      strokeLinecap="round"
                      strokeWidth="3"
                    />
                  </svg>
                </DownIcon>
              </span>
              <span>
                -2,00 %
              </span>
            </span>
            <br />
            <span
              className="subtitle"
            >
              Ano passado
            </span>
          </div>
        </SummaryItem>
        <SummaryItem
          subtitle="Mês passado"
          uppperTitle=""
          value={3}
        >
          <div>
            <span
              className="upper-title"
            />
            <br />
            <span
              className="number"
            >
              <span
                className="up-or-down-icon"
              >
                <UpIcon
                  height="17"
                  viewBox="0 0 16 17"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <svg
                    height="17"
                    viewBox="0 0 16 17"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1.5 8.5l5.793-5.793a1 1 0 0 1 1.414 0L14.5 8.5m-6.5 7v-13"
                      fill="none"
                      stroke="#3C9"
                      strokeLinecap="round"
                      strokeWidth="3"
                    />
                  </svg>
                </UpIcon>
              </span>
              <span>
                3,00 %
              </span>
            </span>
            <br />
            <span
              className="subtitle"
            >
              Mês passado
            </span>
          </div>
        </SummaryItem>
      </div>
    </StyledComponent>
  </BusinessSummary__Wrapper>
</BusinessSummary>
`;
